ucsmsdk/utils/comparesyncmo.py

```diff
@@ -0,0 +1,555 @@
+                                       _MoDiff.ADD_MODIFY))
+
+    diff_with_props = _compare_common_mo(reference_dict, difference_dict,
+                                         include_operational,
+                                         no_version_filter, include_equal,
+                                         exclude_different)
+
+    if diff_with_props:
+        diff_output.extend(diff_with_props)
+
+    return diff_output
+
+
+def sync_ucs_mo(ref_handle, difference,
+                delete_not_present=False,
+                no_version_filter=False):
+    """
+    syncs the difference object on reference ucsm.
+    In other words, make the state of reference ucsm for the respective
+    ref object same as difference object on reference ucsm.
+
+    Args:
+        ref_handle (UcsHandle): connect handle of reference ucsm
+        difference (list of MoDiff objects): output of compare-ucs-mo api
+        delete_not_present (bool): by dafault False. If set to true, will
+                                   delete ref mo which does not exist on
+                                   difference ucsm
+        no_version_filter (bool): by default False: If set to True, ignore
+                                  properties which is introduced in later ucsm
+                                  version than reference ucsm
+
+    Returns:
+        None
+
+    Example:
+        #1
+        ref_mos = [ref_handle.query_dn(dn="org-root/ls-sp")]
+        diff_mos = [diff_handle.query_dn(dn="org-root/ls-sp")]
+        difference = compare_ucs_mo(ref_handle, ref_mos,diff_handle, diff_mos)
+        sync_ucs_mo(ref_handle, difference=difference, delete_not_present=True)
+    """
+
+    if difference is None or not _list_has_values(difference):
+        raise UcsValidationException(
+            "difference object parameter is not provided.")
+
+    to_commit = False
+
+    for mo_diff in difference:
+        mo = mo_diff.input_object
+        class_id = mo.get_class_id()
+
+        if isinstance(mo, GenericMo):
+            UcsWarning("Ignoring '%s'.Unknown ClasId '%s'" % (mo.dn, class_id))
+            continue
+
+        # Remove
+        if mo_diff.side_indicator == _MoDiff.REMOVE and delete_not_present:
+            log.debug("removing mo '%s'." % mo.dn)
+            ref_handle.remove_mo(mo)
+            to_commit = True
+
+        if not no_version_filter:
+            mo_meta = mo.mo_meta
+            if ref_handle.version < mo_meta.version:
+                UcsWarning("Ignoring unsupported class_id '%s' for dn '%s'" %
+                           (class_id, mo.dn))
+                continue
+
+        # Add or Modify
+        if mo_diff.side_indicator == _MoDiff.ADD_MODIFY:
+            add_exists = False
+            if isinstance(mo, GenericMo):
+                mo_meta = None
+            else:
+                mo_meta = mo.mo_meta
+
+            if mo_meta is not None and 'Add' in mo_meta.verbs:
+                add_exists = True
```

Comment:
Given that mo_meta is used only to calculate value of add_exists the above 6 lines can be compressed into 2.
```
If !GMO and 'Add' in mo.mo_meta.verbs:
    add_exists = True
```